app/screens/phone-auth-screen/phone-input.tsx

```diff
@@ -147,17 +145,37 @@ export const PhoneInputScreen: React.FC = () => {
       fetchPolicy: "no-cache",
     })
 
+  useEffect(() => {
+    const getCountryCodeFromIP = async () => {
+      try {
+        const response = (await fetchWithTimeout("https://ipapi.co/json/")) as Response
+        const data = await response.json()
+
+        if (data && data.country_code) {
+          const countryCode = data.country_code
+          setDefaultCode(countryCode)
```

Comment:
You may want to return at line 156 and 160 and then in line 161 you can add a `setDefaultCode` so that it is guaranteed that it is called. Currently if the `if` check in line 154 fails, it will never set the default code.